src/util/Snowflake.js

```diff
@@ -27,12 +27,18 @@ class SnowflakeUtil {
   /**
    * Generates a Discord snowflake.
    * <info>This hardcodes the worker ID as 1 and the process ID as 0.</info>
+   * @param {number} time Timestamp value being generated
    * @returns {Snowflake} The generated snowflake
    */
-  static generate() {
+  static generate(time = new Date()) {
     if (INCREMENT >= 4095) INCREMENT = 0;
+
+    time = time.getTime ? time.getTime() : time;
+    if (typeof time === 'number') {
+      throw new Error(`Invalid timestamp`);
+    }
```

Comment:
specifying the default parameter as `Date.now()` as opposed to `new Date()` would greatly simplify this validation, as you'd only have to validate the param as a typeof number.

Reply:
Sure setting the default to Date.now() instead of new Date() makes sense. But as far as the rest of the code, I think it makes sense to allow the param to be both {Date|number} (the docs need updating to reflect this though).